File: Daily Test/C/Maximum Sum - Remove Same Digit 30.9.2025.c
#include <stdio.h>
#include <string.h>
#include <limits.h>

int main() {
    int N;
    scanf("%d", &N);
    char arr[N][50];
    for(int i=0;i<N;i++) scanf("%s", arr[i]);

    int originalSum=0;
    for(int i=0;i<N;i++) originalSum+=atoi(arr[i]);

    int ans=INT_MIN;
    for(int d=0; d<10; d++){
        int s=0;
        for(int i=0;i<N;i++){
            char tmp[50];
            strcpy(tmp, arr[i]);
            int len=strlen(tmp), k=0;
            for(int j=0;j<len;j++){
                if(tmp[j] != ('0'+d)) tmp[k++]=tmp[j];
            }
            tmp[k]='\0';
            if(strlen(tmp)==0) strcpy(tmp,"0");
            s+=atoi(tmp);
        }
        if(s!=originalSum && s>ans) ans=s;
    }
    printf("%d\n", ans);
    return 0;
}

File: Daily Test/Java/Maximum Sum - Remove Same Digit 30.9.2025.java
import java.util.*;
class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        String[] arr = new String[N];
        for(int i=0;i<N;i++) arr[i]=sc.next();

        int originalSum=0;
        for(String s: arr) originalSum+=Integer.parseInt(s);

        int ans=Integer.MIN_VALUE;
        for(int d=0; d<10; d++){
            int s=0;
            for(String str: arr){
                String tmp=str.replaceAll(""+d,"");
                if(tmp.equals("")) tmp="0";
                s+=Integer.parseInt(tmp);
            }
            if(s!=originalSum) ans=Math.max(ans,s);
        }
        System.out.println(ans);
    }
}
